src/m365/aad/commands/app/app-add.ts

```diff
@@ -441,18 +441,54 @@ class AadAppAddCommand extends GraphCommand {
     delete v2Manifest.id;
     delete v2Manifest.appId;
     delete v2Manifest.publisherDomain;
+    
     // extract secrets from the manifest. Store them in a separate variable
-    // and remove them from the manifest because we need to create them
-    // separately
-    const secrets: { name: string, expirationDate: Date }[] = this.getSecretsFromManifest(v2Manifest);
+    let secrets: { name: string, expirationDate: Date }[] = this.getSecretsFromManifest(v2Manifest);
+
+    if (args.options.withSecret) {
+      secrets = [];
```

Comment:
If the manifest has a secret already, wouldn't it be clearer to reuse it, rather than remove it and create a random one? The desired outcome is to create an AAD app with a secret so if there's one or more already in the manifest, we could just as well ignore the `withSecret` option altogether.

Reply:
good point 🤙